### PR TITLE
Migration product api

### PR DESCRIPTION
**Pull request by coconutperm**

### CODE REVIEW

File: server/order/router/product_router.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+});
+
+router.get('/', (request, response) => {
+  product.getProductList()
+    .then((productList) => {
+      response.send(productList);
+    });
+});
+
+module.exports = router;
```
Comment (romandev) [inline server/order/router/product_router.js]: We don't use module.exports. Please use ECMA version export.

File: server/order/router/product_router.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import express from 'express';
+import * as product from '../controller/product_controller';
+
+const router = express.router();
+
+router.post('/', (request, response) => {
```
Comment (romandev) [inline server/order/router/product_router.js]: Let's make a new style router.

File: server/order/router/product.router.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  get(request, response) {
+    product.getProductList()
+      .then((productList) => {
+        response.send(productList);
+      });
+  }
+
+  post(request, response) {
+    if (request.body.name && request.body.amount) {
+      product.addProduct(JSON.parse(JSON.stringify(request.body)))
```
Comment (romandev) [inline server/order/router/product.router.js]: Why are you calling stringify() and then parse() again?

File: server/order/controller/product_controller.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import {Product} from '../model/product';
+
+/**
+ * Add product data what admin added.
+ * @param {json} productData data what admin sent.
+ * @return {promise} result product data what admin sent.
+ */
+export function addProduct(productData) {
+  return new Promise((resolve, reject) => {
+    Product.findOneAndUpdate({name: productData.name}
```
Comment (romandev) [inline server/order/controller/product_controller.js]: As far as I know, the mongoose already supports Promise as follows:
  - http://mongoosejs.com/docs/promises.html

File: server/order/controller/product_controller.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  });
+}
+
+/**
+ * Get product data list to saved.
+ * @return {promise} result data list of product
+ */
+export function getProductList() {
+  return new Promise((resolve, reject) => {
+    Product.find({}, (error, productList) => {
```
Comment (romandev) [inline server/order/controller/product_controller.js]: ditto

File: server/order/controller/product_controller.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      if (productList) {
+        resolve(JSON.stringify(productList));
+      } else {
+        reject(-1);
```
Comment (romandev) [inline server/order/controller/product_controller.js]: This is not right way.
We should define Error object with error message explicitly.

File: server/order/router/product.router.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Application from '../../base/application.js';
+import * as product from '../controller/product_controller';
+
+@Application.route('/product')
```
Comment (romandev) [inline server/order/router/product.router.js]: Add jsdoc-style comments.

File: server/order/router/product.router.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  post(request, response) {
+    if (request.body.name && request.body.amount) {
+      product.addProduct(JSON.parse(JSON.stringify(request.body)))
+        .then((product) => {
+          response.send(product);
+        });
+    } else {
+      response.sendStatus(400);
```
Comment (romandev) [inline server/order/router/product.router.js]: Please use named constant.

Comment (coconutperm) [inline server/order/router/product.router.js]: The error code must be formally defined.

File: server/order/controller/product_controller.js (hunk 5 above)
Comment (coconutperm) [reply server/order/controller/product_controller.js]: Mongoose queries are not promises. However, they do have a .then() function for yield and async/await. If you need a fully-fledged promise, use the .exec() function.

File: server/order/controller/product_controller.js (hunk 4 above)
Comment (coconutperm) [reply server/order/controller/product_controller.js]: Mongoose queries are not promises. However, they do have a .then() function for yield and async/await. If you need a fully-fledged promise, use the .exec() function.

Comment (romandev) [reply server/order/controller/product_controller.js]: Yeah, but you can just use then() or exec() instead.
IMHO, wrapper seems unnecessary, do you agree with me?

**Pull request closed without merging**